### PR TITLE
EVG-17480 Update Slack API calls 

### PR DESCRIPTION
**Pull request by malikchaya2**
[EVG-17480](https://jira.mongodb.org/browse/EVG-17480)

### Description 
When available, use the slackMemberId and not the userName. 

### Testing 
Tested on staging and the slack sandbox.

### CODE REVIEW

File: model/notification/notification.go [hunk 1]
```diff
@@ -260,6 +267,22 @@ func (n *Notification) SetTaskMetadata(ID string, execution int) {
...
 }
 
+func FormatSlackTarget(target string) (string, error) {
+	//check if it's a userID (can potentially also be a user-group)
+	if strings.HasPrefix(target, "@") {
+		//use the memberId if it exists
+		trimmedTarget := strings.TrimPrefix(target, "@")
+		user, err := user.FindBySlackUsername(trimmedTarget)
+		if err != nil {
+			return "", errors.Wrapf(err, "finding user by slack username '%s'", target)
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/notification/notification.go:277]: Good point, I'll instead log and return.

File: model/notification/notification.go [hunk 2]
```diff
@@ -260,6 +267,22 @@ func (n *Notification) SetTaskMetadata(ID string, execution int) {
 	n.Metadata.TaskExecution = execution
 }
 
+func FormatSlackTarget(target string) (string, error) {
+	//check if it's a userID (can potentially also be a user-group)
+	if strings.HasPrefix(target, "@") {
+		//use the memberId if it exists
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/notification/notification.go:273]: removed

File: model/notification/notification.go [hunk 3]
```diff
@@ -260,6 +267,22 @@ func (n *Notification) SetTaskMetadata(ID string, execution int) {
 	n.Metadata.TaskExecution = execution
 }
 
+func FormatSlackTarget(target string) (string, error) {
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/notification/notification.go:270]: done

File: model/notification/notification.go [hunk 4]
```diff
@@ -260,6 +267,22 @@ func (n *Notification) SetTaskMetadata(ID string, execution int) {
...
+	//check if it's a userID (can potentially also be a user-group)
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply model/notification/notification.go:271]: removed

File: rest/route/notification.go [hunk 5]
```diff
@@ -200,10 +201,15 @@ func (h *slackNotificationPostHandler) Run(ctx context.Context) gimlet.Responder
 	for _, a := range h.APISlack.Attachments {
 		attachments = append(attachments, a.ToService())
 	}
+	// this should be the memberId
```
[the remark this answers is not in the document]
Comment (malikchaya2) [reply rest/route/notification.go:204]: removed

File: model/notification/notification.go [hunk 6]
```diff
@@ -267,14 +267,17 @@ func (n *Notification) SetTaskMetadata(ID string, execution int) {
...
 func FormatSlackTarget(target string) (string, error) {
-	//check if it's a userID (can potentially also be a user-group)
 	if strings.HasPrefix(target, "@") {
-		//use the memberId if it exists
 		trimmedTarget := strings.TrimPrefix(target, "@")
 		user, err := user.FindBySlackUsername(trimmedTarget)
 		if err != nil {
-			return "", errors.Wrapf(err, "finding user by slack username '%s'", target)
+			grip.Error(message.WrapError(err, message.Fields{
+				"message": "finding user by slack username",
```
Comment (Kimchelly) [inline model/notification/notification.go:277]: nit message format: `"message": "could not find user by Slack username"`
Can we also mention we're falling back to the default due to this error?

**Review by Kimchelly — APPROVED** [review]
LGTM mod nit.

**Pull request merged**